commands/build-image.ts

```diff
@@ -56,25 +57,7 @@ export async function buildImage(actionContext: IActionContext, dockerFileUri: v
     const defaultContextPath = configOptions.get('imageBuildContextPath', '');
     let dockerFileItem: Item | undefined;
 
-    let rootFolder: vscode.WorkspaceFolder;
-    if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length === 1) {
-        rootFolder = vscode.workspace.workspaceFolders[0];
-    } else {
-        let selected = await vscode.window.showWorkspaceFolderPick();
-        if (!selected) {
-            throw new UserCancelledError();
-        }
-        rootFolder = selected;
-    }
-
-    if (!rootFolder) {
-        if (!vscode.workspace.workspaceFolders) {
-            vscode.window.showErrorMessage('Docker files can only be built if VS Code is opened on a folder.');
-        } else {
-            vscode.window.showErrorMessage('Docker files can only be built if a workspace folder is picked in VS Code.');
-        }
-        return;
-    }
+    let rootFolder: vscode.WorkspaceFolder = await quickPickWorkspaceFolder('To build Docker files you must first open a folder or workspace in VS Code.');
```

Comment:
I see `you must first open a folder or workspace in VS Code.` being repeated in all 3 calls. Possibly add that to the function itself?

Reply:
I'd rather not. Doing string concatenation of parts of sentences causes problems if we want to localize later.

Reply:
Interesting. I thought the loc could happen in the function, at the point we throw the error. 
I don't feel strongly about this, I was just wondering if there was a way to force consistency in the messages in the future. These errors would stand out to the user more than to they would to the dev.

Reply:
I assumed you mean something like:
   `To ${message} you need to first open...`

That tends to be problematic.  I agree this is non-optimal, but I haven't found a better way, except to restrict the message to be more generic. I think this is good enough, though.  Thx.